app/widget/marker/marker.cpp

```diff
@@ -0,0 +1,100 @@
+/***
+
+  Olive - Non-Linear Video Editor
+  Copyright (C) 2021 Olive Team
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+***/
+
+#include "marker.h"
+
+#include <QPainter>
+
+#include "common/qtutils.h"
+#include "ui/colorcoding.h"
+#include "widget/menu/menu.h"
+#include "widget/menu/menushared.h"
+
+namespace olive {
+
+Marker::Marker(QWidget *parent) :
+	QWidget(parent),
+    marker_color_(Qt::green)
+{
+  setSizePolicy(QSizePolicy::Fixed, QSizePolicy::Fixed);
+  setMinimumSize(20, 20);
+  setContextMenuPolicy(Qt::CustomContextMenu);
+
+  connect(this, &Marker::customContextMenuRequested, this, &Marker::ShowContextMenu);
+
+  color_coding_menu_ = new ColorLabelMenu();
```

Comment:
Not giving this menu a parent will create a memory leak. I'd be more inclined to make this object on the stack in `ShowContextMenu` and connect it there too just out of ease.

Reply:
Could you check I've done this right?

Reply:
The new `ShowContextMenu`? Seems good to me